sample-resource-plugin/src/main/java/org/opensearch/sample/resource/actions/transport/ShareResourceTransportAction.java

```diff
@@ -33,12 +33,13 @@
  */
 public class ShareResourceTransportAction extends HandledTransportAction<ShareResourceRequest, ShareResourceResponse> {
     private static final Logger log = LogManager.getLogger(ShareResourceTransportAction.class);
-    private final ResourceSharingClient resourceSharingClient;
+
+    @Inject(optional = true)
+    public SampleResourceExtension sampleResourceExtension;
```

Comment:
Nice to see that field injection is working...its interesting how in OpenSearch' fork of Guice from 10 years ago that fields must be public. In newer versions of Guice, that's not the case but until OpenSearch removes the fork of guice and adds a dependency on Guice instead we are stuck with our fork.

Reply:
Yes, if I have some time on my hands I'll look into raising a PR to remove Guice implementation and declare dependency on the package..